### PR TITLE
CVSB-139 - implement add-defect and unit tests

### PR DESCRIPTION
**Pull request by danieliordache**

### CODE REVIEW

File: src/pages/testing/defects/add-defect-category/add-defect-category.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    this.vehicleTest = navParams.get('test');
+  }
+
+  ngOnInit() {
+    this.getDefectCategories();
+  }
+
+  getDefectCategories(): void {
+    if (this.defectCategories == null) {
+      // this.defectCategoryService.getDefectCategories().subscribe(
```
Comment (acozianu) [inline src/pages/testing/defects/add-defect-category/add-defect-category.ts]: remove comments

File: src/pages/testing/defects/add-defect-item/add-defect-item.scss [hunk 2]
```diff
@@ -0,0 +1,22 @@
+  ion-app.ios page-add-defect-item {
+    [padding] .scroll-content {
+      padding: 0;
+    }
+
+    .item-ios {
+      ion-icon[item-end] {
+        color: #c8c7cc;
```
Comment (acozianu) [inline src/pages/testing/defects/add-defect-item/add-defect-item.scss]: use variable for color

File: src/pages/testing/defects/add-defect-item/add-defect-item.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  constructor(public navCtrl: NavController, public navParams: NavParams) {
+    this.vehicleTest = navParams.get('test');
+    this.defectCategories = navParams.get('defectCategoryTaxonomy');
+    this.parentDefectCategory = navParams.get('parentDefectCategory');
+    this.parentDefectCategoryId = navParams.get('parentDefectCategoryId');
+    this.additionalInfo = navParams.get('additionalInfo');
+    this.forVehicleType = navParams.get('forVehicleType');
+  }
+
+  // addDefect(defect: DefectModel): void {
```
Comment (acozianu) [inline src/pages/testing/defects/add-defect-item/add-defect-item.ts]: remove comments

File: src/pages/testing/defects/advisory-details/advisory-details.scss [hunk 4]
```diff
@@ -0,0 +1,20 @@
+page-advisory-details {
+  ion-content {
+    .scroll-content {
+      background-color: #f4f4f4;
```
Comment (acozianu) [inline src/pages/testing/defects/advisory-details/advisory-details.scss]: use variables for color

Comment (danieliordache) [reply src/pages/testing/defects/advisory-details/advisory-details.scss]: fixed

File: src/pages/testing/defects/add-defect-item/add-defect-item.ts (hunk 3 above)
Comment (danieliordache) [reply src/pages/testing/defects/add-defect-item/add-defect-item.ts]: fixed

File: src/pages/testing/defects/add-defect-item/add-defect-item.scss (hunk 2 above)
Comment (danieliordache) [reply src/pages/testing/defects/add-defect-item/add-defect-item.scss]: fixed

File: src/pages/testing/defects/add-defect-category/add-defect-category.ts (hunk 1 above)
Comment (danieliordache) [reply src/pages/testing/defects/add-defect-category/add-defect-category.ts]: fixed

**Pull request merged**